### PR TITLE
[remove units] add partial_eval_jaxpr_nounits

### PR DESCRIPTION
**Pull request by mattjj**
No callers yet, but this is to be used for initial-style higher-order primitives (HOPs, like in lax/control_flow.py) to remove units. Follow-up PRs will switch the HOPs one-by-one to use this new function.

### CODE REVIEW

File: jax/interpreters/partial_eval.py [hunk 1]
```diff
@@ -785,88 +798,65 @@ def convert_envvars_to_constvars(jaxpr: Jaxpr, num_env_vars: int) -> Jaxpr:
 def _split_aval(unknown: bool, aval: AbstractValue) -> Tuple[AbstractValue, AbstractValue]:
   return (abstract_unit, aval) if unknown else (aval, abstract_unit)
 
-def partial_eval_jaxpr(jaxpr: ClosedJaxpr, unknowns: Sequence[bool],
-                       instantiate: Union[bool, Sequence[bool]],
-                       ) -> Tuple[ClosedJaxpr, ClosedJaxpr, Sequence[bool]]:
-  """Specializes a Jaxpr given an indication of which inputs are known.
```
Comment (froystig) [inline jax/interpreters/partial_eval.py:791]: Keep (some of) this docstring around?

**Review by froystig — APPROVED** [review]

File: jax/interpreters/partial_eval.py (hunk 1 above)
Comment (mattjj) [reply jax/interpreters/partial_eval.py:791]: Well, I'd rather not document the deprecated one, and most of it is incorrect for the new one. I can add a docstring for the new function though...

Comment (mattjj) [reply jax/interpreters/partial_eval.py:791]: Done!

Comment (froystig) [reply jax/interpreters/partial_eval.py:791]: Nice. I've found it to be one of the more useful comments in this file, because of the example.

Comment (mattjj) [reply jax/interpreters/partial_eval.py:791]: Good suggestion.

**Pull request merged**